test/IRGen/witness_method_default.swift

```diff
@@ -0,0 +1,23 @@
+// RUN: %target-swift-frontend %s -emit-ir | %FileCheck --check-prefix=CHECK %s -DINT=i%target-ptrsize
+
+public protocol DummyProtocol { }
+
+public protocol SIMDStorageStub {
+ associatedtype Scalar : DummyProtocol
+}
+
+public protocol SIMDScalarStub {
+ associatedtype SIMD2Storage : SIMDStorageStub
+   where SIMD2Storage.Scalar == Self
```

Comment:
In this example, Self conforms to DummyProtocol via this requirement. Either getConformsTo() on Self should drop DummyProtocol, since its implied by the conformance to SIMDScalarStub, or we should not allow this where clause at all.

I'm leaning toward the latter, since name lookup won't currently find members of DummyProtocol on a type parameter constrained to SIMDScalarStub.

Reply:
@slavapestov how would you write `protocol SIMDScalar` in SIMDVector.swift?

Reply:
/cc @stephentyrone

Reply:
I agree with @slavapestov that we should complain about the addition of a conformance requirement to `Self` through this same-type constraint, although we might have to stage it in as a warning. That doesn't let us avoid fixing this code path, though, which is more wrong than I'd originally realized.

Reply:
@atrick in this test case, you would write `protocol SIMDScalarStub : DummyProtocol`